### PR TITLE
Android: don't require deprecated tools

### LINKED ISSUE #21735: Building OpenCV for Android using Android cmdline-tools
Android cmdline-tools no longer installs the deprecated Android SDK Tools package. This causes a cmake configure error when OpenCV tries to find these tools in `${ANDROID_SDK}/tools` if they're not already installed.

##### System information (version)

- OpenCV => :grey_question:
- Operating System / Platform => :grey_question:
- Compiler => :grey_question:

##### Steps to reproduce

1. Clean install Android SDK using the latest `cmdline-tools` package.
2. Build OpenCV for Android using the included python helper script, e.g. `python3 /opencv/platforms/android/build_sdk.py --sdk_path=/opt/android-sdk --no_samples_build --config=/opencv/platforms/android/ndk-22.config.py`
Build fails during configure step with a message like this one:
```
-- Update variable ANDROID_SDK_ROOT from environment: /opt/android-sdk
-- Update variable ANDROID_HOME from environment: /opt/android-sdk
-- Update variable ANDROID_SDK from environment: /opt/android-sdk
CMake Error at cmake/android/OpenCVDetectAndroidSDK.cmake:64 (message):
  Android SDK Tools: can't automatically find Android SDK Tools.  Specify
  path via ANDROID_SDK_TOOLS variable
Call Stack (most recent call first):
  cmake/android/OpenCVDetectAndroidSDK.cmake:173 (ocv_detect_android_sdk_tools)
  CMakeLists.txt:746 (include)


-- Configuring incomplete, errors occurred!
See also "/build/o4a/CMakeFiles/CMakeOutput.log".
See also "/build/o4a/CMakeFiles/CMakeError.log".
Traceback (most recent call last):
  File "/opencv/platforms/android/build_sdk.py", line 445, in <module>
    builder.build_library(abi, do_install)
  File "/opencv/platforms/android/build_sdk.py", line 262, in build_library
    execute(cmd)
  File "/opencv/platforms/android/build_sdk.py", line 30, in execute
    raise Fail("Child returned: %s" % retcode)
__main__.Fail: Child returned: 1
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found any solution
   
 - [x] I updated to the latest OpenCV version and the issue is still there
   
 - [ ] There is reproducer code and related data files: videos, images, onnx, etc

### PR DESCRIPTION
**Pull request by anders-nylander**
Checking for these deprecated is no longer necessary, and infact broken on fresh Android SDK installs. Remove the check.

resolves #21735

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders_only=Android armeabi-v7a,docs
```

### CODE REVIEW

**Comment by alalek** [comment]
Patch should not introduce regressions of used configurations.

"Android armeabi-v7a" build has the following setup:

```
--   Android NDK:                   /opt/android/android-ndk-r18b (ver 18.1.5063045)
--     Android ABI:                 armeabi-v7a
--     NDK toolchain:               arm-linux-androideabi-clang
--     STL type:                    c++_static
--     Native API level:            21
--   Android SDK:                   /opt/android/android-sdk.gradle (tools: 26.1.1 build tools: 28.0.3)
```

(all of these dependencies are likely from 2018-2019)

---

It would be nice if you could take a look how to support both configurations.

File: cmake/android/OpenCVDetectAndroidSDK.cmake [hunk 1]
```diff
@@ -56,8 +56,13 @@ endmacro()
 macro(ocv_detect_android_sdk_tools)
   # https://developer.android.com/studio/releases/sdk-tools.html
   if(NOT DEFINED ANDROID_SDK_TOOLS)
-    if(DEFINED ANDROID_SDK AND EXISTS "${ANDROID_SDK}/tools")
-      set(ANDROID_SDK_TOOLS "${ANDROID_SDK}/tools" CACHE INTERNAL "Android SDK Tools path")
+    if(DEFINED ANDROID_SDK)
+      if (EXISTS "${ANDROID_SDK}/cmdline-tools")
```
Comment (opencv-alalek) [inline cmake/android/OpenCVDetectAndroidSDK.cmake:60]: They are not the same and can't be exchanged. Just compare:

cmdline-tools/latest/source.properties:
```
Pkg.Revision=7.0
Pkg.Path=cmdline-tools;7.0
Pkg.Desc=Android SDK Command-line Tools
```

vs

tools/source.properties
```
Pkg.UserSrc=false
Pkg.Revision=26.1.1
Platform.MinPlatformToolsRev=20
Pkg.Dependencies=emulator
Pkg.Path=tools
Pkg.Desc=Android SDK Tools
```

---

OpenCV doesn't use "tools" with modern "gradle" flow actually. They are use with legacy "ant" build flow.

---

Try to comment out `#ocv_detect_android_sdk_tools()` call.

Comment (dkurt) [reply cmake/android/OpenCVDetectAndroidSDK.cmake]: Replaced to a simple check that tools found.

`cmdline-tools` tools changed their versioning so the latest one is 9.0. More about my build steps here: https://github.com/opencv/opencv/pull/23642

File: cmake/android/OpenCVDetectAndroidSDK.cmake [hunk 2]
```diff
@@ -173,12 +173,12 @@ if(BUILD_ANDROID_PROJECTS)
   ocv_detect_android_sdk_tools()
   ocv_detect_android_sdk_build_tools()
 
-  if(ANDROID_SDK_TOOLS_VERSION VERSION_LESS 14)
+  if(DEFINED ANDROID_SDK_TOOLS_VERSION AND ANDROID_SDK_TOOLS_VERSION VERSION_LESS 14)
```
Comment (opencv-alalek) [inline cmake/android/OpenCVDetectAndroidSDK.cmake:176]: Please clean CMake cache and ensure that configuration process doesn't stop on the line 64.

Comment (dkurt) [reply cmake/android/OpenCVDetectAndroidSDK.cmake:176]: Thanks, sorry for confusion. My sdk contained both `tools` and `cmdline-tools`. Needed reinstall.

**Pull request merged**